src/ship.js

```diff
@@ -0,0 +1,28 @@
+const Itinerary = require("./itinerary");
+class Ship {
+  constructor(itinerary) {
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+  }
+
+  setSail() {
+    const itinerary = this.itinerary;
+    const currentPortInd = itinerary.ports.indexOf(this.currentPort);
+    if (currentPortInd === this.itinerary.ports.length - 1) {
+      throw new Error("You have reached the end of the itinerary");
+    }
+    this.currentPort.removeShip(this);
+    this.previousPort = this.currentPort;
+    this.currentPort = null;
+  }
+
+  dock() {
+    const newInd = this.itinerary.ports.indexOf(this.previousPort) + 1;
+    this.currentPort = this.itinerary.ports[newInd];
+    this.currentPort.addShip(this);
```

Comment:
you access `this.itinerary.ports` multiple times in this function. You could assign this to a variable to make your code more DRY i.e.
```
        const ports = this.itinerary.ports;
        const newIndex = ports.indexOf(this.previousPort) + 1;

        this.currentPort = ports[newIndex];
        this.currentPort.addShip(this);
```